tests/e2e/gc_block_volume_expansion.go

```diff
@@ -1334,6 +1334,309 @@ var _ = ginkgo.Describe("[csi-guest] Volume Expansion Test", func() {
+		newSize1 := currentPvcSize.DeepCopy()
+		newSize1.Add(resource.MustParse("1Gi"))
+		newSize2 := currentPvcSize.DeepCopy()
+		newSize2.Add(resource.MustParse("2Gi"))
+		newSize3 := currentPvcSize.DeepCopy()
+		newSize3.Add(resource.MustParse("8Gi"))
+
+		var wg sync.WaitGroup
+		wg.Add(3)
+		go resize(client, pvclaim, currentPvcSize, newSize2, &wg)
+		go resize(client, pvclaim, currentPvcSize, newSize1, &wg)
+		go resize(client, pvclaim, currentPvcSize, newSize3, &wg)
+		wg.Wait()
+
+		pvclaim, err = client.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvclaim.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+
+		pvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(newSize3) != 0 {
+			framework.Failf("error updating pvc size %q", pvclaim.Name)
+		}
+
+		ginkgo.By("Checking for PVC request size change on SVC PVC")
+		b, err := verifyPvcRequestedSizeUpdateInSupervisorWithWait(svcPVCName, newSize3)
+		gomega.Expect(b).To(gomega.BeTrue())
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Waiting for controller volume resize to finish")
+		err = waitForPvResizeForGivenPvc(pvclaim, client, totalResizeWaitPeriod)
+		framework.ExpectNoError(err, "While waiting for pvc resize to finish")
+
+		ginkgo.By("Checking for resize on SVC PV")
+		verifyPVSizeinSupervisor(svcPVCName, newSize3)
+
+		ginkgo.By("Waiting for file system resize to finish")
+		pvclaim, err = waitForFSResize(pvclaim, client)
+		framework.ExpectNoError(err, "while waiting for fs resize to finish")
+
+		pvcConditions := pvclaim.Status.Conditions
+		expectEqual(len(pvcConditions), 0, "pvc should not have conditions")
+
+		ginkgo.By("Verify filesystem size for mount point /mnt/volume1")
+		fsSize, err := getFSSizeMb(f, pod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("FileSystemSize after PVC resize %d mb , FileSystemSize Before PVC resize %d mb ", fsSize, originalFsSize)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		// Filesystem size may be smaller than the size of the block volume
+		// so here we are checking if the new filesystem size is greater than
+		// the original volume size as the filesystem is formatted for the
+		// first time
+		gomega.Expect(fsSize).Should(gomega.BeNumerically(">", 7000/(1024*1024)), fmt.Sprintf("error updating filesystem size for %q. Filesystem size is not more than 7GB", pvclaim.Name))
+
+		ginkgo.By("File system resize finished successfully")
+		pvcsize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+		actualSizeOfPVC := sizeInMb(pvcsize.DeepCopy())
+		expectedSize := int64(10240)
+		framework.Logf("Actual size after converting to MB : %d", actualSizeOfPVC)
+		if actualSizeOfPVC != expectedSize {
+			framework.Failf("Received wrong disk size after volume expansion. Expected: %d Actual: %d", expectedSize, actualSizeOfPVC)
+
+		}
```

Comment:
this is not required, I feel. It should already be covered in the checks we have before.